public/src/managers/ServerProvider.ts

```diff
@@ -81,114 +81,116 @@ export class ServerProvider implements IUserProvider, ICourseProvider {
 
     public async getAllUserEnrollments(userID: number): Promise<Enrollment[]> {
         const result = await this.grpcHelper.getEnrollmentsByUser(userID);
-        if (result.status.getCode() !== 0 || !result.data) {
+        if (!this.responseCodeSuccess(result) || !result.data) {
             return [];
         }
         return result.data.getEnrollmentsList();
     }
 
     public async getUsersForCourse(
         course: Course, noGroupMembers?: boolean,
-        state?: Enrollment.UserStatus[]): Promise<Enrollment[]> {
+        status?: Enrollment.UserStatus[]): Promise<Enrollment[]> {
 
-        const result = await this.grpcHelper.getEnrollmentsByCourse(course.getId(), noGroupMembers, state);
-        if (result.status.getCode() !== 0 || !result.data) {
+        const result = await this.grpcHelper.getEnrollmentsByCourse(course.getId(), noGroupMembers, status);
+        if (!this.responseCodeSuccess(result) || !result.data) {
             return [];
         }
         return result.data.getEnrollmentsList();
     }
 
     public async getAssignments(courseID: number): Promise<Assignment[]> {
         const result = await this.grpcHelper.getAssignments(courseID);
-        if (result.status.getCode() !== 0 || !result.data) {
+        if (!this.responseCodeSuccess(result) || !result.data) {
             return [];
         }
         return result.data.getAssignmentsList();
     }
 
     public async addUserToCourse(course: Course, user: User): Promise<boolean> {
         const result = await this.grpcHelper.createEnrollment(course.getId(), user.getId());
-        return result.status.getCode() === 0;
+        return this.responseCodeSuccess(result);
     }
 
-    public async changeUserState(link: Enrollment, state: Enrollment.UserStatus): Promise<Status> {
-        link.setStatus(state);
-        const result = await this.grpcHelper.updateEnrollment(link);
+    public async changeUserStatus(enrollment: Enrollment, status: Enrollment.UserStatus): Promise<Status> {
+        const originalStatus = enrollment.getStatus();
+        enrollment.setStatus(status);
+        const result = await this.grpcHelper.updateEnrollment(enrollment);
+        if (!this.responseCodeSuccess(result)) {
+            enrollment.setStatus(originalStatus);
+        }
         return result.status;
     }
 
     public async approveAll(courseID: number): Promise<boolean> {
         const result = await this.grpcHelper.updateEnrollments(courseID);
-        if (result.status.getCode() !== 0 || !result.data) {
-            return false;
-        }
-        return true;
+        return result.data ? this.responseCodeSuccess(result) : false;
```

Comment:
This pattern is different from the others below. Suggest to harmonize these. See more below.

Reply:
```typescript
return this.isSuccess(result);
```
(note `isSuccess()` is only one of several names that I mix and match for the same method...)